File: src/main/java/com/evanlennick/webserver/mimetypes/MimeType.java
package com.evanlennick.webserver.mimetypes;

import java.util.Arrays;
import java.util.List;

public enum MimeType {

    IMG_PNG("image/png", Arrays.asList("png", "x-png")),

    IMG_JPG("image/jpeg", Arrays.asList("jpg", "jpeg", "jpe", "jfif", "jfif-tbnl")),

    TEXT_PLAIN("text/plain; charset=UTF-8", Arrays.asList("txt", "text", "c", "c++", "cc", "conf", "java", "log", "pl")),

    TEXT_HTML("text/html; charset=utf-8", Arrays.asList("html", "htm", "shtml", "htmls", "htx", "acgi"));

    private String mimeTypeString;

    private List<String> extensions;

    MimeType(String mimeTypeString, List<String> extensions) {
        this.mimeTypeString = mimeTypeString;
        this.extensions = extensions;
    }

    public String getMimeTypeString() {
        return mimeTypeString;
    }

    public List<String> getExtensions() {
        return extensions;
    }
}

File: src/main/java/com/evanlennick/webserver/mimetype/MimeType.kt
package com.evanlennick.webserver.mimetype

import java.util.Arrays

enum class MimeType private constructor(val mimeTypeString: String, val extensions: List<String>) {

    IMG_PNG("image/png", Arrays.asList("png", "x-png")),

    IMG_JPG("image/jpeg", Arrays.asList("jpg", "jpeg", "jpe", "jfif", "jfif-tbnl")),

    IMG_ICO("image/x-icon", Arrays.asList("ico")),

    APPLICATION_JSON("application/json", Arrays.asList("json")),

    APPLICATION_XML("application/xml", Arrays.asList("xml")),

    APPLICATION_JAVASCRIPT("application/javascript", Arrays.asList("js")),

    TEXT_PLAIN("text/plain; charset=UTF-8", Arrays.asList("txt", "text", "c", "c++", "cc", "conf", "java", "log", "pl")),

    TEXT_HTML("text/html; charset=utf-8", Arrays.asList("html", "htm", "shtml", "htmls", "htx", "acgi"))
}
